skxray/core/correlation.py

```diff
@@ -402,3 +351,366 @@ def auto_corr_scat_factor(lags, beta, relaxation_rate, baseline=1):
+                                  num_pixels, img_per_level, lag_steps,
+                                  current_img_time, level=level,
+                                  buf_no=cur[level]-1)
+                level += 1
+
+                # Checking whether there is next level for processing
+                processing = level < num_levels
+
+    for q in range(np.max(labels)):
+        x0 = two_time[:, :, q]
+        two_time[:, :, q] = (np.tril(x0) + np.tril(x0).T
+                               - np.diag(np.diag(x0)))
+    # Two time correlation processing time
+    logger.info("Two Time Correlation - Processing time for {0} images took"
+                " {1} seconds." .format(n, (time.time() - start_time)))
+
+    return two_time
+
+
+def _two_time_process(buf, two_time, label_array, num_bufs, num_pixels,
+                      img_per_level, lag_steps, current_img_time, level,
+                      buf_no):
+    """
+    Parameters
+    ----------
+    buf: array
+        image data array to use for two time correlation
+    two_time: array
+        two time correlation matrix
+    label_array: array
+        Elements not inside any ROI are zero; elements inside each
+        ROI are 1, 2, 3, etc. corresponding to the order they are specified
+        in edges and segments
+    num_bufs: int, even
+        number of buffers(channels)
+    num_pixels : array
+        number of pixels in certain roi's
+        roi's, dimensions are : [number of roi's]
+    img_per_level: array
+        to track how many images processed in each level
+    lag_steps : array
+        delay or lag steps for the multiple tau analysis
+        shape num_levels
+    current_img_time : int
+        the current image number
+    level : int
+        the current multi-tau level
+    buf_no : int
+        the current buffer number
+    """
+    img_per_level[level] += 1
+
+    # in multi-tau correlation other than first level all other levels
+    #  have to do the half of the correlation
+    if level == 0:
+        i_min = 0
+    else:
+        i_min = num_bufs//2
+
+    for i in range(i_min, min(img_per_level[level], num_bufs)):
+        (t_index, tmp_binned, pi_binned,
+         fi_binned) = _help_process(level, num_bufs, buf_no, i, buf,
+                                    label_array)
+
+        tind1 = (current_img_time - 1)
+
+        tind2 = (current_img_time - lag_steps[t_index] - 1)
+
+        if not isinstance(current_img_time, int):
+            nshift = 2**(level-1)
+            for i in range(-nshift+1, nshift+1):
+                two_time[int(tind1+i),
+                           int(tind2+i)] = (tmp_binned/(pi_binned *
+                                                        fi_binned))*num_pixels
+        else:
+            two_time[tind1, tind2] = tmp_binned/(pi_binned *
+                                                   fi_binned)*num_pixels
+
+
+def _help_process(level, num_bufs, buf_no, i, buf, label_array):
```

Comment:
Only checking my e-mails still having jet lag, wake up around 4.30am (Sri Lankan time), better than last week ( 2.30am) So nothing to do until others wake up. Sure leave comments , I will fix after I come back,